flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -385,24 +386,38 @@ public void testInputOutputFormat() {
 	@Test
 	public void testShuffleMode() {
 		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
-		// fromElements -> Map -> Print, will not chain since the batch data exchange mode
-		DataStream<Integer> mapDataStream = env.fromElements(1, 2, 3)
-			.map((MapFunction<Integer, Integer>) value -> value).setParallelism(2);
+		// fromElements -> Filter -> Map -> Print
```

Comment:
Usually, I like to split these tests up, so that each test checks one isolated thing. The structure of each test would be roughly:
```
void testSomething() {
   //setup
  
   //action

   //verification
}
```

Otherwise, the whole test has to be touched whenever we change something for one of the scheduling modes or the scheduling logic.

Reply:
Good point!